app/src/state/actions/workshop-actions.ts

```
@@ -65,3 +66,101 @@ export const fetchWorkshopFilesForDownload = action(async (workshop: SharedWorks
+
+  if (!response.ok) {
+    let errorCode = ErrorCodes.SERVER_RESPONSE_ERROR;
+    switch (response.status) {
+      case 404:
+        errorCode = ErrorCodes.WORKSHOP_FILE_NOT_FOUND_FOR_DOWNLOAD;
+        break;
+      case 401:
+        errorCode = ErrorCodes.NOT_LOGGED_IN;
+        break;
+      case 500:
+        errorCode = ErrorCodes.SERVER_RESPONSE_ERROR;
+        break;
+    }
+    displayServerError(response, errorCode, ErrorMessages[errorCode]);
+    return;
+  }
+
+  const file = await response.blob();
+  console.log(response);
+  if (file) {
+    try {
+      await downloadFile(file, fileName);
+    } catch (error) {
+      pushError(ErrorCodes.UNABLE_TO_READ_FILE_OR_EMPTY_FILE);
+    }
+  }
+});
+
+export const fetchWorkshopDetailsDatapack = action(async (datapackTitle: string, workshop: SharedWorkshop) => {
+  const requestURL = `/workshop/workshop-datapack/${workshop.workshopId}/${datapackTitle}`;
+
+  const recaptchaToken = await getRecaptchaToken("fetchWorkshopDataPackForDownload");
+  if (!recaptchaToken) return null;
+
+  const response = await fetcher(requestURL, {
+    method: "GET",
+    credentials: "include",
+    headers: {
+      "recaptcha-token": recaptchaToken
+    }
+  });
+
+  console.log("Response from fetchWorkshopDetailsDatapack:", response);
+
+  if (!response.ok) {
+    console.log(response);
+    let errorCode = ErrorCodes.SERVER_RESPONSE_ERROR;
+    switch (response.status) {
+      case 404:
+        errorCode = ErrorCodes.WORKSHOP_FILE_NOT_FOUND_FOR_DOWNLOAD;
+        break;
+      case 401:
+        errorCode = ErrorCodes.NOT_LOGGED_IN;
+        break;
+      case 500:
+        errorCode = ErrorCodes.SERVER_RESPONSE_ERROR;
+        break;
+    }
+    displayServerError(response, errorCode, ErrorMessages[errorCode]);
+    return;
+  }
+
+  const data = await response.json();
+  const { fileName, fileData, fileType } = data;
+
+  // Convert base64 to Blob
+  const byteCharacters = atob(fileData);
+  const byteArrays = [];
+
```

with `shared` responses that come from the server to the app, we probably wanna have an asserts statement that is defined in the `index.ts` file in the `shared` module that we call here to make sure the types are correct, if the response is incorrect the asserts should throw the error and propogate to the try catch i asked for above in the comments